### PR TITLE
Feature/ds4-42 Setup foundation for recipes

### PR DESCRIPTION
**Pull request by MagnusHJensen**
Setups foundation for `commonrecipe` and `recipe` modules.

It also adds the required Interface and model files inside the common recipe.

### CODE REVIEW

File: backend/application/pom.xml
Comment (Siole) [inline backend/application/pom.xml:1]: modules uses Pascal in the class diagram

File: backend/commonrecipe/pom.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
```
Comment (Siole) [inline backend/commonrecipe/pom.xml:44]: Is the plugin needed? Asking coz I'm not sure

File: backend/commonrecipe/src/main/java/dk/sdu/controllab/commonrecipe/CreateRecipeRequest.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package dk.sdu.controllab.commonrecipe;
+
+import java.util.List;
+import java.util.UUID;
+
+public record CreateRecipeRequest(List<UUID> inputIds, UUID outputId) {
```
Comment (Siole) [inline backend/commonrecipe/src/main/java/dk/sdu/controllab/commonrecipe/CreateRecipeRequest.java:6]: Doesn't match the class diagram

File: backend/pom.xml [hunk 3]
```diff
@@ -18,6 +18,8 @@
     <modules>
         <module>application</module>
         <module>common</module>
+        <module>commonrecipe</module>
```
Comment (Siole) [inline backend/pom.xml:21]: Pascal

File: backend/recipe/pom.xml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
```
Comment (Siole) [inline backend/recipe/pom.xml:47]: Plugin needed?

File: backend/commonrecipe/src/main/java/dk/sdu/controllab/commonrecipe/UpdateRecipeRequest.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package dk.sdu.controllab.commonrecipe;
+
+import java.util.List;
+import java.util.UUID;
+
+public record UpdateRecipeRequest(List<UUID> inputIds, UUID outputId) {
```
Comment (Siole) [inline backend/commonrecipe/src/main/java/dk/sdu/controllab/commonrecipe/UpdateRecipeRequest.java:6]: Doesn't match class diagram

File: backend/CommonRecipe/src/main/java/dk/sdu/controllab/commonrecipe/CreateRecipeRequest.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package dk.sdu.controllab.commonrecipe;
+
+import java.util.List;
+import java.util.UUID;
+
+public record CreateRecipeRequest(List<UUID> inputItemIds, UUID outputItemId) {
```
Comment (Siole) [inline backend/CommonRecipe/src/main/java/dk/sdu/controllab/commonrecipe/CreateRecipeRequest.java:6]: Hvorfor er det her en record og ikke en @getter?

File: backend/CommonRecipe/src/main/java/dk/sdu/controllab/commonrecipe/UpdateRecipeRequest.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package dk.sdu.controllab.commonrecipe;
+
+import java.util.List;
+import java.util.UUID;
+
+public record UpdateRecipeRequest(List<UUID> inputItemIds, UUID outputItemId) {
```
Comment (Siole) [inline backend/CommonRecipe/src/main/java/dk/sdu/controllab/commonrecipe/UpdateRecipeRequest.java:6]: no Spring annotation?

**Review by Siole — CHANGES_REQUESTED** [review]
Tag stilling til record.

File: backend/CommonRecipe/src/main/java/dk/sdu/controllab/commonrecipe/UpdateRecipeRequest.java (hunk 7 above)
Comment (MagnusHJensen) [reply backend/CommonRecipe/src/main/java/dk/sdu/controllab/commonrecipe/UpdateRecipeRequest.java:6]: Nope, det her er bare en POJO (Plain Old Java Object) klasse.

**Review by Siole — APPROVED** [review]

**Pull request merged**